R/hms.R

```diff
@@ -274,3 +274,41 @@ print.hms <- function(x, ...) {
   cat(format(x), sep = "\n")
   invisible(x)
 }
+
+# Sequence generation -----------------
+
+#' @rdname hms
+#' @inheritParams base::seq
+#' @export
+seq.hms <- function(
+    from = hms(1),
+    to = hms(1),
+    by = NULL,
+    ...
+) {
+
+  if (!is_hms(to)) {
+    abort(sprintf(
+      "`to` isn't of class `hms` (current class: `%s`).",
+      class(to)[1]
+    ))
+  }
+  
+  from <- vec_cast(from, numeric())
+  to <- vec_cast(to, numeric())
+
+  if (!is.null(by)) {
+    if (!(is_hms(by) || inherits(by, "difftime"))) {
+      abort(sprintf(
+        "`by` isn't of class `hms` or `difftime` (current class: `%s`).",
+        class(by)[1]
+      ))
+    }
+    # if by is difftime, we can't use vec_cast on it directly
+    if (!is_hms(by)) by <- as_hms(by)
+    by <- vec_cast(by, numeric())
```

Comment:
Only saw this now. I read it like this:

- We can't cast a numeric to a difftime
- We want to cast everything to a difftime anyway

Why is the comment and the special case important then?

```suggestion
    by <- vec_cast(as_hms(by), numeric())
```